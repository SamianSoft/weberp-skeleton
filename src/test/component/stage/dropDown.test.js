import React from 'react';
import { mount, shallow, render } from 'enzyme';
import DropDownComponent from '../../../component/Stage/DropDown';


describe('StageComponent', () => {
    it('should render correctly with no props', () => {
      const component = shallow(<DropDownComponent />);
    
      expect(component).toMatchSnapshot();
    });

    it('should render stage items correctly with given data', () => {
        const testData = [
            {
                key: 'test1',
                title: 'test1',
                priority: 1,
                tickWhenPass: true,
                color: '#FDFDFD',
            },
            {
                key: 'test2',
                title: 'test2',
                priority: 2,
                tickWhenPass: false,
                color: '#000',
            },
            {
                key: 'test3',
                title: 'test3',
                priority: 3,
                tickWhenPass: true,
                color: 'green',
            },
        ];
        const component = shallow(<DropDownComponent data={testData} />);
        expect(component).toMatchSnapshot();
      });
  });