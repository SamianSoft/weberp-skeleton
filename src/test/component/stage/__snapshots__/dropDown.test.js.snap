// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StageComponent should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropDown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "attachment": "top center",
      "constraints": Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ],
      "renderElement": [Function],
      "renderElementTag": "div",
      "renderElementTo": null,
      "renderTarget": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "attachment": "top center",
        "constraints": Array [
          Object {
            "attachment": "together",
            "to": "window",
          },
        ],
        "renderElement": [Function],
        "renderElementTag": "div",
        "renderElementTo": null,
        "renderTarget": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`StageComponent should render stage items correctly with given data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropDown
    data={
      Array [
        Object {
          "color": "#FDFDFD",
          "key": "test1",
          "priority": 1,
          "tickWhenPass": true,
          "title": "test1",
        },
        Object {
          "color": "#000",
          "key": "test2",
          "priority": 2,
          "tickWhenPass": false,
          "title": "test2",
        },
        Object {
          "color": "green",
          "key": "test3",
          "priority": 3,
          "tickWhenPass": true,
          "title": "test3",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "attachment": "top center",
      "constraints": Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ],
      "renderElement": [Function],
      "renderElementTag": "div",
      "renderElementTo": null,
      "renderTarget": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "attachment": "top center",
        "constraints": Array [
          Object {
            "attachment": "together",
            "to": "window",
          },
        ],
        "renderElement": [Function],
        "renderElementTag": "div",
        "renderElementTo": null,
        "renderTarget": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
